### PR TITLE
feat: generate code deprecation tags

### PR DESCRIPTION
**Pull request by jkaster**
- adds `(DEPRECATED)` to deprecated method parameters that don't include the text "deprecated"
- adds deprecated attributes for relevant Kotlin, Swift, Go, and C# methods
- minor edits to readme
- updated examples index
- smoke tests for Kotlin and Swift pass

### CODE REVIEW

**Review by joeldodge79 — COMMENTED** [review]
Did you intentionally not regenerate TS-SDK and PY-SDK? Also, I didn't open `examplesIndex.json` but it looks like a big change so just making sure that was intentional.

The python and codegen stuff LGTM! But I don't feel qualified to review on the other languages or apix. Happy to approve if someone else can speak for those

**Review by josephaxisa — COMMENTED** [review]
API Explorer related changes lgtm. Happy to approve once the questions around the typescript and python SDKs are answered.

**Comment by jkaster** [comment]
All SDKs have been updated where appropriate and regenerated. As mentioned in the PR description, the examples index is also updated. That should be something that runs as part of CI/CD or as part of the release process. And also whenever new examples are submitted. TODO

**Comment by joeldodge79** [comment]
I'm confused about why the generated csharp, go, kotlin, and swift SDKs all seem to have the `deprecated` stuff but typescript and python don't - even though I see the required changes in their generators. I assume this is still WIP @jkaster ?

**Comment by jkaster** [comment]
I'll have to look at it again, because the deprecated tags should be there.

**Comment by jkaster** [comment]
> I'm confused about why the generated csharp, go, kotlin, and swift SDKs all seem to have the `deprecated` stuff but typescript and python don't - even though I see the required changes in their generators. I assume this is still WIP @jkaster ?

@joeldodge79 that's because the specification with deprecated parameters won't be available until Looker 22.4 is released. This change added deprecated params to the test specs used for testing the generator. This PR is done after adding *new* deprecation tags for some of the generators, and has new unit tests that verify tagging deprecated params as well as method deprecation attributes where appropriate. TypeScript already had `@deprecated` tags for methods.

File: packages/sdk-codegen/src/kotlin.gen.ts [hunk 1]
```diff
@@ -291,12 +291,20 @@ ${this.commentHeader(
   indent,
   this.methodHeaderComment(method, streamer)
 ).trimEnd()}
-${indent}${this.jvmOverloads(method)}fun ${method.name}(
+${this.deprecated(indent, method)}${indent}${this.jvmOverloads(method)}fun ${
+      method.name
+    }(
```
Comment (joeldodge79) [inline packages/sdk-codegen/src/kotlin.gen.ts:296]: This change was the source of my confusion. It's not related to deprecated method params - instead it's back-filling already existing functionality to deprecate a whole method. Typescript already had it and you added it in this PR kotlin, csharp, go, swift but not python.

When I saw `deprecated` showing up in generated code for some languages and not others I got really confused.

I'm ok with this as is, but I feel like had this bullet point in your PR description been a different PR I would have gotten through this one a lot quicker.

`adds deprecated attributes for relevant Kotlin, Swift, Go, and C# methods`

**Review by joeldodge79 — APPROVED** [review]

**Pull request merged**